baseplate/clients/kombu.py

```diff
@@ -11,18 +12,42 @@
 from kombu import Connection
 from kombu import Exchange
 from kombu.pools import Producers
+from prometheus_client import Counter
+from prometheus_client import Histogram
 from thrift import TSerialization
 from thrift.protocol.TBinaryProtocol import TBinaryProtocolAcceleratedFactory
 from thrift.protocol.TProtocol import TProtocolFactory
+from vine.promises import promise  # type: ignore
 
 from baseplate import Span
 from baseplate.clients import ContextFactory
 from baseplate.lib import config
+from baseplate.lib.prometheus_metrics import default_latency_buckets
 from baseplate.lib.secrets import SecretsStore
 
 
 T = TypeVar("T")
 
+amqp_producer_labels = [
+    "amqp_address",
+    "amqp_virtual_host",
+    "amqp_exchange",  # exchange_type
+    "amqp_queue",  # exchange_name
+    "amqp_success",
+]
+
+PROCESSING_TIME = Histogram(
```

Comment:
Should this be `AMQP_PROCESSING_TIME`, as Kombu can be used for more than just reading from AMQP queues?